### PR TITLE
Add external logs provider support

### PR DESCRIPTION
**Pull request by eddycharly**
# Changes

This PR adds support for an external logs provider.
This external logs provider is used as a fallback when the logs are not available anymore from k8s directly.

It allows displaying logs in the Dashboard when the pods baking TaskRuns have been deleted.

It includes:
- a new flag in the backend to configure the logs service url
- a fallback mechanism in the frontend to fetch logs from the configured url when the logs cannot be fetched from k8s
- a walkthrough to show an example on how the setup logs persistence

Walkthrough visible here https://github.com/eddycharly/dashboard/blob/external-logs/docs/walkthrough/walkthrough-logs.md

/cc @a-roberts @AlanGreene 

# Submitter Checklist

These are the criteria that every PR should meet, please check them off as you
review them:

- [ ] Includes [tests](https://github.com/tektoncd/community/blob/master/standards.md#principles) (if functionality changed/added)
- [x] Includes [docs](https://github.com/tektoncd/community/blob/master/standards.md#principles) (if user facing)
- [ ] Commit messages follow [commit message best practices](https://github.com/tektoncd/community/blob/master/standards.md#commit-messages)

_See [the contribution guide](https://github.com/tektoncd/dashboard/blob/master/CONTRIBUTING.md)
for more details._

### CODE REVIEW

**Comment by eddycharly** [comment]
Rebased quickly, will need to get back to it.
From the top of my head, TODO:
- use a proxy to fetch logs from the backend instead of frontend

File: src/containers/PipelineRun/PipelineRun.js [hunk 1]
```diff
@@ -247,6 +247,7 @@ function mapStateToProps(state, ownProps) {
       getPipelineRunsErrorMessage(state) ||
       getTasksErrorMessage(state) ||
       getTaskRunsErrorMessage(state),
+    externalLogsURL: externalLogsURL(state),
```
Comment (AlanGreene) [inline src/containers/PipelineRun/PipelineRun.js:250]:
```suggestion
    externalLogsURL: getExternalLogsURL(state),
```

File: src/containers/PipelineRun/PipelineRun.js [hunk 2]
```diff
@@ -21,6 +21,7 @@ import { Link } from 'react-router-dom';
 import { injectIntl } from 'react-intl';
 
 import {
+  externalLogsURL,
```
Comment (AlanGreene) [inline src/containers/PipelineRun/PipelineRun.js:24]:
```suggestion
  getExternalLogsURL,
```

File: docs/walkthrough/walkthrough-logs.md [hunk 3]
```diff
@@ -0,0 +1,355 @@
...
+  hosts:
+    - minio.127.0.0.1.nip.io
+
+resources:
+  requests:
+    memory: 100M
+EOF
+```
+
+The deployed instance will use hard coded access and secret keys `ACCESSKEY` / `SECRETKEY` and the service will be exposed externally at `http://minio.127.0.0.1.nip.io`.
```
[the remark this answers is not in the document]
Comment (eddycharly) [reply docs/walkthrough/walkthrough-logs.md:66]: The walkthrough uses harcoded `ACCESSKEY` / `SECRETKEY` string literals. (those are configured when deploying minio helm chart)

File: docs/walkthrough/walkthrough-logs.md [hunk 4]
```diff
@@ -0,0 +1,355 @@
...
+                )
+              )
+              .then(() => response.end());
+          }
+          app.options('*', cors());
+          app.get('/logs/:namespace/:pod/:container', cors(), (req, res) => streamLogs(req.params.namespace, req.params.pod, req.params.container, res));
+          app.listen(3000, '0.0.0.0');
+          EOF
+
+          npm install aws-sdk express cors
```
[the remark this answers is not in the document]
Comment (eddycharly) [reply docs/walkthrough/walkthrough-logs.md]: True, fixed.

**Comment by AlanGreene** [comment]
@eddycharly there's an NPM issue that's intermittently affecting installs... https://status.npmjs.org/incidents/cksjqc1w11v5

Hopefully it's resolved soon.

**Pull request merged**